src/components/transcript/transcript.tsx

```diff
@@ -1,494 +1,379 @@
+      });
+      if (event.type !== 'click') {
+        this._skipTranscriptButtonRef?.focus();
+      }
     }
+  };
 
-    private _createEngine = () => {
-        const { captions } = this.props;
-        if (!captions || captions.length === 0) {
-            this._engine = null;
-            return;
-        }
-        this._engine = new CuepointEngine<CaptionItem>(captions);
-        this._syncVisibleTranscript();
-    };
-
-    private _syncVisibleTranscript = (forceSnapshot = false) => {
-        const { currentTime } = this.props;
-        this.setState((state: TranscriptState) => {
-            if (!this._engine) {
-                return {
-                    highlightedMap: {}
-                };
-            }
-
-            const transcriptUpdate = this._engine.updateTime(currentTime, forceSnapshot);
-            if (transcriptUpdate.snapshot) {
-                this._silence = false;
-                const highlightedMap = transcriptUpdate.snapshot.reduce((acc, item) => {
-                    return { ...acc, [item.id]: true };
-                }, {});
-                return {
-                    highlightedMap
-                };
-            }
-
-            if (!transcriptUpdate.delta) {
-                return state;
-            }
-
-            const { show, hide } = transcriptUpdate.delta;
-
-            if (show.length === 0 && hide.length > 0) {
-                this._silence = true;
-                return state;
-            }
-
-            if (show.length > 0 || hide.length > 0) {
-                let newHighlightedMap: Record<number, true> = {};
-                if (this._silence) {
-                    this._silence = false;
-                } else {
-                    newHighlightedMap = { ...state.highlightedMap };
-                }
-                show.forEach((caption: CaptionItem) => {
-                    newHighlightedMap[caption.id] = true;
-                });
-
-                hide.forEach((caption: CaptionItem) => {
-                    delete newHighlightedMap[caption.id];
-                });
-
-                return {
-                    highlightedMap: newHighlightedMap
-                };
-            }
-            return state;
-        });
-    };
-
-    private _enableAutoScroll = (event: any) => {
-        event.preventDefault();
-        if (this.state.isAutoScrollEnabled) {
-          return;
-        }
-        if (
-            event.type === "click" ||
-            event.keyCode === KeyboardKeys.Space ||
-            event.keyCode === KeyboardKeys.Enter
-        ) {
-            this._preventScrollEvent = true;
-            this.setState({
-                isAutoScrollEnabled: true
-            });
-            if (event.type !== "click") {
-                this._skipTranscriptButtonRef?.focus();
-            }
+  private _renderScrollToButton = () => {
+    const {isAutoScrollEnabled} = this.state;
+    return (
+      <A11yWrapper onClick={this._enableAutoScroll}>
+        <div
```

Comment:
missed Icon in autoscroll button:
<img width="810" alt="image" src="https://user-images.githubusercontent.com/51074448/181214499-55f22846-34d0-4ead-8794-dbb355b3ceda.png">